src/api/java/com/minecolonies/api/colony/requestsystem/StandardRequestManager.java

```
@@ -0,0 +1,1269 @@
+
+    /**
+     * Map that holds the class that resolver can resolve. Used during lookup.
+     */
+    @NotNull
+    private final HashMap<Class, Collection<IRequestResolver>> requestClassResolverMap = new HashMap<>();
+    /**
+     * The fallback resolver used to resolve directly to the player.
+     * TODO: Assign resolver once implemented.
+     */
+    @NotNull
+    private final IRequestResolver                             playerResolver          = null;
+    /**
+     * Colony of the manager.
+     */
+    @NotNull
+    private IColony colony;
+
+    public StandardRequestManager(@NotNull IColony colony)
+    {
+        this.colony = colony;
+    }
+
+    /**
+     * The colony this manager manages the requests for.
+     *
+     * @return The colony this manager manages the requests for.
+     */
+    @NotNull
+    @Override
+    public IColony getColony()
+    {
+        return colony;
+    }
+
+    /**
+     * Method used to serialize the current request system to NBT.
+     *
+     * @return The NBTData that describes the current request system
+     */
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        return null;
+    }
+
+    /**
+     * Class used to handle the inner workings of the request system with regards to providers.
+     */
+    private final static class ProviderHandler
+    {
+
+        /**
+         * Method used to get the registered resolvers for a given provider.
+         *
+         * @param manager  The manager to pull the data from.
+         * @param provider The provider you are requesting the resolvers for.
+         * @return The registered resolvers that belong to the given provider.
+         *
+         * @throws IllegalArgumentException when the token is not belonging to a registered provider.
+         */
+        private static ImmutableCollection<IToken> getRegisteredResolvers(StandardRequestManager manager, IRequestResolverProvider provider) throws IllegalArgumentException
```

missing finals here

---

Should we really throw the exception here? or catch it and return an empty collection?

---

Never ever return something that might indicate a logical correct state if it is not. If somebody is requesting with an invalid token that is there fault. The method contract clearly states that it throws the exception, and as such the caller knows that It can throw it, it is clearly documented and it is up to the caller to handle the fact of an illegal token.